tests/components/august/test_config_flow.py

```diff
@@ -0,0 +1,195 @@
+"""Test the August config flow."""
+from asynctest import patch
+from august.authenticator import ValidationResult
+
+from homeassistant import config_entries, setup
+from homeassistant.components.august.const import (
+    CONF_ACCESS_TOKEN_CACHE_FILE,
+    CONF_INSTALL_ID,
+    CONF_LOGIN_METHOD,
+    DOMAIN,
+    VERIFICATION_CODE_KEY,
+)
+from homeassistant.components.august.exceptions import (
+    CannotConnect,
+    InvalidAuth,
+    RequireValidation,
+)
+from homeassistant.const import CONF_PASSWORD, CONF_TIMEOUT, CONF_USERNAME
+
+
+async def test_form(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.august.config_flow.AugustGateway.authenticate",
+        return_value=True,
+    ), patch(
+        "homeassistant.components.august.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.august.async_setup_entry", return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_LOGIN_METHOD: "email",
+                CONF_USERNAME: "my@email.tld",
+                CONF_PASSWORD: "test-password",
+            },
+        )
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "my@email.tld"
+    assert result2["data"] == {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5068548)